src/main/webapp/table.js

```diff
@@ -0,0 +1,149 @@
+/* Class representing Table, used for sorting and rendering.*/
+//TODO: Check which additional functions need to be bound in constructor
+class Table {
+  constructor(dataTable, name, colSchemas, id) {
+    this.name = name;
+    this.colSchemas = colSchemas;
+    this.id = id;
+    this.dataTable = new Array(dataTable.length); // Number rows
+    this.sortDirections = new Array(colSchemas.length); // Number cols
+    this.makeTableWithTypes(dataTable);
+
+    this.setTable = this.setTable.bind(this);
+  }
+
+  getName() {
+    return this.name;
+  }
+
+  getDataTable() {
+    return this.dataTable;
+  }
+
+  getDataType(colIndex) {
+    const colSchema = this.colSchemas[colIndex];
+    return colSchema.schemaType;
+  }
+
+  // Initialize table with actual column types
+  makeTableWithTypes(dataTable) {
+    let rowIndex = 0;
+    for (rowIndex in dataTable) {
+      const row = dataTable[rowIndex];
+      this.dataTable[rowIndex] = new Array(row.length);
+  
+      let col;
+      for (col in row) {
+        const cell = row[col];
+        const type = this.getDataType(col);
+        if (cell == "NULL") {
+          this.dataTable[rowIndex][col] = "";
+        } else {
+          if (type == "INT64") {
+            const cellToInt = parseInt(cell);
+            this.dataTable[rowIndex][col] = cellToInt;
+          } else {
+            this.dataTable[rowIndex][col] = cell;
+          }
+        }
+      }
+    }
+  }
+
+  fetchTable() {
+    setTimeout(function() {
```

Comment:
what's the purpose of this setTimeout?

Reply:
I got rid of it, it wasn't actually needed